src/frontend/screens/ManualGpsScreen/DdForm.js

```diff
@@ -0,0 +1,247 @@
+// @flow
+import * as React from "react";
+import { View, TextInput, StyleSheet } from "react-native";
+import { defineMessages, FormattedMessage } from "react-intl";
+
+import Text from "../../sharedComponents/Text";
+import Select from "../../sharedComponents/Select";
+import { BLACK, LIGHT_GREY } from "../../lib/styles";
+import {
+  type FormProps,
+  POSITIVE_DECIMAL_REGEX,
+  getInitialCardinality,
+} from "./shared";
+
+const MAX_COORDINATE_INPUT_LENGTH = 11;
+
+const m = defineMessages({
+  invalidCoordinates: {
+    id: "screens.ManualGpsScreen.DdForm.invalidCoordinates",
+    defaultMessage: "Invalid coordinates",
+  },
+  latitude: {
+    id: "screens.ManualGpsScreen.DdForm.latitude",
+    defaultMessage: "Latitude",
+  },
+  longitude: {
+    id: "screens.ManualGpsScreen.DdForm.longitude",
+    defaultMessage: "Longitude",
+  },
+  north: {
+    id: "screens.ManualGpsScreen.DdForm.north",
+    defaultMessage: "North",
+  },
+  south: {
+    id: "screens.ManualGpsScreen.DdForm.south",
+    defaultMessage: "South",
+  },
+  east: {
+    id: "screens.ManualGpsScreen.DdForm.east",
+    defaultMessage: "East",
+  },
+  west: {
+    id: "screens.ManualGpsScreen.DdForm.west",
+    defaultMessage: "West",
+  },
+});
+
+const DIRECTION_OPTIONS_NORTH_SOUTH = [
+  {
+    value: "N",
+    label: m.north.defaultMessage,
+  },
+  {
+    value: "S",
+    label: m.south.defaultMessage,
+  },
+];
+
+const DIRECTION_OPTIONS_EAST_WEST = [
+  {
+    value: "E",
+    label: m.east.defaultMessage,
+  },
+  {
+    value: "W",
+    label: m.west.defaultMessage,
+  },
+];
```

Comment:
Can translations be generally used like this?

Reply:
No, because when you change the app language at runtime then this will not change. Best to define these constants inside the component, and then use:

```js
const { formatMessage } = useIntl()
```

And then for the label use:

```js
{
  value: "E"
  label: formatMessage(m.east)
}
```

This will, at runtime, look up the language and look up the translated value, with the fallback to the `defaultMessage` if no translation is defined (or app is in English).